views.go

```diff
@@ -391,6 +397,70 @@ func (lk *logKeeper) viewAllLogs(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func (lk *logKeeper) viewTestInDatabase(r *http.Request, buildID string, testID string) (*logFetchResponse, *apiError) {
+	build, err := model.FindBuildById(buildID)
+	if err != nil || build == nil {
+		return nil, &apiError{Err: "view test by id: build not found", code: http.StatusNotFound}
+	}
+
+	test, err := model.FindTestByID(testID)
+	if err != nil || test == nil {
+		return nil, &apiError{Err: "test not found"}
+	}
+
+	logsChan, err := model.MergedTestLogs(test)
+	if err != nil {
+		lk.logErrorf(r, "Error finding global logs during test: %v", err)
+		return nil, &apiError{Err: err.Error(), code: http.StatusInternalServerError}
+	}
+
+	result := logFetchResponse{
+		logLines: logsChan,
+		build:    build,
+		test:     test,
+	}
+	return &result, nil
+}
+
+func (lk *logKeeper) viewTestInS3(r *http.Request, buildID string, testID string) (*logFetchResponse, *apiError) {
+	build, err := lk.opts.Bucket.FindBuildByID(r.Context(), buildID)
+	if err != nil || build == nil {
+		return nil, &apiError{Err: "view test in S3 by id: build not found", code: http.StatusNotFound}
```

Comment:
If `err != nil` I'd think we'd want to return like a 500 or something.

Reply:
This is the behavior from the original code, but I think it makes sense to go ahead and update it to return the right thing.

Reply:
This is fixed in the parallelized version.